### PR TITLE
Show last set DHW Setpoint if OTGW-interface returns no value (0.0)

### PR DESCRIPTION
**Pull request by RobKuipers**

### CODE REVIEW

File: hardware/OTGWBase.cpp [hunk 1]
```diff
@@ -399,7 +400,7 @@ void OTGWBase::ParseLine()
 		_status.Return_water_temperature = static_cast<float>(atof(results[idx++].c_str()));				SendTempSensor(idx - 1, 255, _status.Return_water_temperature, "Return Water Temperature");
 		_status.DHW_setpoint_boundaries=results[idx++];
 		_status.Max_CH_setpoint_boundaries=results[idx++];
-		_status.DHW_setpoint = static_cast<float>(atof(results[idx++].c_str()));							UpdateSetPointSensor(idx - 1, _status.DHW_setpoint, "DHW Setpoint");
+		_status.DHW_setpoint = static_cast<float>(atof(results[idx++].c_str())); if(_status.DHW_setpoint!=0.0f)							UpdateSetPointSensor(idx - 1, _status.DHW_setpoint, "DHW Setpoint");
```
Comment (gizmocuz) [inline hardware/OTGWBase.cpp]: This if should be on a separate line please
Maybe this patch needs to be implemented for the others as well (return water, control setpoint, boiler water)

File: hardware/OTGWBase.cpp [hunk 2]
```diff
@@ -302,6 +302,7 @@ void OTGWBase::SetSetpoint(const int idx, const float temp)
 		_log.Log(LOG_STATUS, "OTGW: Setting Heating SetPoint to: %.1f", temp);
 		sprintf(szCmd, "SW=%.1f\r\n", temp);
 		WriteInt((const unsigned char*)&szCmd, (const unsigned char)strlen(szCmd));
+                UpdateSetPointSensor(idx, temp, "DHW Setpoint");
```
Comment (gizmocuz) [inline hardware/OTGWBase.cpp]: this should be updated automatically in the next poll... but if you can not wait that long, you want to do this for the other set points as well ?

File: hardware/OTGWBase.cpp [hunk 3]
```diff
@@ -281,35 +281,30 @@ void OTGWBase::SendOutsideTemperature()
...
 	char szCmd[30];
-	if (idx == 1)
-	{
-		//Control Set Point (MsgID=1)
-		_log.Log(LOG_STATUS, "OTGW: Setting Control SetPoint to: %.1f", temp);
-		sprintf(szCmd, "CS=%.1f\r\n", temp);
-		WriteInt((const unsigned char*)&szCmd, (const unsigned char)strlen(szCmd));
-	}
-	else if (idx == 5)
+	if (idx == 5)
```
Comment (gizmocuz) [inline hardware/OTGWBase.cpp]: Why is setpoint 1 removed ?

File: hardware/OTGWBase.cpp [hunk 4]
```diff
@@ -281,35 +281,30 @@ void OTGWBase::SendOutsideTemperature()
...
+		UpdateSetPointSensor(idx, temp, "Room Setpoint");
 	}
 	else if (idx == 15)
 	{
 		//DHW setpoint (MsgID=56)
 		_log.Log(LOG_STATUS, "OTGW: Setting Heating SetPoint to: %.1f", temp);
 		sprintf(szCmd, "SW=%.1f\r\n", temp);
 		WriteInt((const unsigned char*)&szCmd, (const unsigned char)strlen(szCmd));
-                UpdateSetPointSensor(idx, temp, "DHW Setpoint");
+        UpdateSetPointSensor(idx, temp, "DHW Setpoint");
```
Comment (gizmocuz) [inline hardware/OTGWBase.cpp]: missing ident here, why is this removed?

File: hardware/OTGWBase.cpp [hunk 5]
```diff
@@ -400,8 +395,10 @@ void OTGWBase::ParseLine()
 		_status.Return_water_temperature = static_cast<float>(atof(results[idx++].c_str()));				SendTempSensor(idx - 1, 255, _status.Return_water_temperature, "Return Water Temperature");
 		_status.DHW_setpoint_boundaries=results[idx++];
 		_status.Max_CH_setpoint_boundaries=results[idx++];
-		_status.DHW_setpoint = static_cast<float>(atof(results[idx++].c_str())); if(_status.DHW_setpoint!=0.0f)							UpdateSetPointSensor(idx - 1, _status.DHW_setpoint, "DHW Setpoint");
-		_status.Max_CH_water_setpoint = static_cast<float>(atof(results[idx++].c_str()));					UpdateSetPointSensor(idx - 1, _status.Max_CH_water_setpoint, "Max_CH Water Setpoint");
+		_status.DHW_setpoint = static_cast<float>(atof(results[idx++].c_str())); 
+		if(_status.DHW_setpoint!=0.0f) UpdateSetPointSensor(idx - 1, _status.DHW_setpoint, "DHW Setpoint");
```
Comment (gizmocuz) [inline hardware/OTGWBase.cpp]: Please follow the default coding standards
if (1==1)
<ident>do something

or 

if (1==1)
{
<ident>do something
}

**Review by gizmocuz — COMMENTED** [review]
Sorry, a few more reviews

**Comment by RobKuipers** [comment]
Hi,


See inline.


Thanks for the review. Will check back soon.

Regards,

Rob


________________________________
Van: gizmocuz <notifications@github.com>
Verzonden: donderdag 22 december 2016 14:27
Aan: domoticz/domoticz
CC: RobKuipers; Author
Onderwerp: Re: [domoticz/domoticz] Show last set DHW Setpoint if OTGW-interface returns no value (0.0) (#1065)


@gizmocuz commented on this pull request.

Sorry, a few more reviews

________________________________

In hardware/OTGWBase.cpp<https://github.com/domoticz/domoticz/pull/1065#pullrequestreview-14153149>:

> @@ -281,35 +281,30 @@ void OTGWBase::SendOutsideTemperature()
 void OTGWBase::SetSetpoint(const int idx, const float temp)
 {
        char szCmd[30];
-       if (idx == 1)
-       {
-               //Control Set Point (MsgID=1)
-               _log.Log(LOG_STATUS, "OTGW: Setting Control SetPoint to: %.1f", temp);
-               sprintf(szCmd, "CS=%.1f\r\n", temp);
-               WriteInt((const unsigned char*)&szCmd, (const unsigned char)strlen(szCmd));
-       }
-       else if (idx == 5)
+       if (idx == 5)


Why is setpoint 1 removed ?

Because Control Set Point is a thermometer, it cannot be adjusted by the user (from Domoticz). OTGW allows setting CS, but then you have to -know-  that you have to put it back to zero to unset it. I think this is why someone (wisely) has made this a thermometer instead of a thermostat. I follow his lead by removing this unreachable code.

________________________________

In hardware/OTGWBase.cpp<https://github.com/domoticz/domoticz/pull/1065#pullrequestreview-14153149>:

>       }
        else if (idx == 15)
        {
                //DHW setpoint (MsgID=56)
                _log.Log(LOG_STATUS, "OTGW: Setting Heating SetPoint to: %.1f", temp);
                sprintf(szCmd, "SW=%.1f\r\n", temp);
                WriteInt((const unsigned char*)&szCmd, (const unsigned char)strlen(szCmd));
-                UpdateSetPointSensor(idx, temp, "DHW Setpoint");
+        UpdateSetPointSensor(idx, temp, "DHW Setpoint");


missing ident here, why is this removed?

Oops, I'm using vi on my raspberrypi. I'll check again.

________________________________

In hardware/OTGWBase.cpp<https://github.com/domoticz/domoticz/pull/1065#pullrequestreview-14153149>:

> @@ -400,8 +395,10 @@ void OTGWBase::ParseLine()
                _status.Return_water_temperature = static_cast<float>(atof(results[idx++].c_str()));                            SendTempSensor(idx - 1, 255, _status.Return_water_temperature, "Return Water Temperature");
                _status.DHW_setpoint_boundaries=results[idx++];
                _status.Max_CH_setpoint_boundaries=results[idx++];
-               _status.DHW_setpoint = static_cast<float>(atof(results[idx++].c_str())); if(_status.DHW_setpoint!=0.0f)                                                 UpdateSetPointSensor(idx - 1, _status.DHW_setpoint, "DHW Setpoint");
-               _status.Max_CH_water_setpoint = static_cast<float>(atof(results[idx++].c_str()));                                       UpdateSetPointSensor(idx - 1, _status.Max_CH_water_setpoint, "Max_CH Water Setpoint");
+               _status.DHW_setpoint = static_cast<float>(atof(results[idx++].c_str()));
+               if(_status.DHW_setpoint!=0.0f) UpdateSetPointSensor(idx - 1, _status.DHW_setpoint, "DHW Setpoint");


Please follow the default coding standards
if (1==1)
do something

or

if (1==1)
{
do something
}

Check! My bad. Will do!



-

**Comment by RobKuipers** [comment]
Hi Gizmocuz,

I made a new attempt. This time using VS. Hopefully this looks better. Thanks for your patience.

Rob

**Comment by gizmocuz** [comment]
Okey, but even if the setpoint with idx can not be called, i would prefer to leave this in.
As original author of this, i had probably a good reason to do so
The rest of the patch looks perfect now!

**Comment by RobKuipers** [comment]
Check. I’ll reverse that change!

Back in a minute.

**Comment by gizmocuz** [comment]
Thanks
